### DISCUSSION TITLE
Project change recovery functionality

### DISCUSSION THREAD

**Discussion by ottop** [godotengine/godot-proposals#5588] [Editor]
Godot does not seem to save edited files in a project automatically by default. If you force quit the application, there is no prompt to attempt recovering lost data either, at least on the 3.5.1 version. 

This means that if the software crashes during the development of a project, any unsaved progress is lost. It is quite common in software nowadays to provide some mechanism to attempt recovering lost data, so having such a feature or defaulting to an autosave mechanism would make sense.

**Comment by Zireael07** [comment]
Did you try 'Save on focus loss' setting as a workaround?

**Reply by Calinou** [reply]
There's also an autosave setting you can enable in the Editor Settings, but I think it only affects scripts, not scenes.

The issue for enabling autosave by default is that it doesn't save to a temporary folder, so it can cause changes to be overwritten when you don't want to. Also, saving large scenes can take a while, during which the editor becomes unusable. This is made worse by [having a progress bar always show when the scene is saving, even if it takes a short time](https://github.com/godotengine/godot-proposals/issues/4196).
